lib/src/core/engine.dart

```diff
@@ -433,9 +373,19 @@ class Engine extends Disposable with EventsEmittable<EngineEvent> {
       final isPrimaryOrPublisher = event.isPrimary ||
           (_hasPublished && event is EnginePublisherPeerStateUpdatedEvent);
 
-      if (isPrimaryOrPublisher && event.state.isDisconnectedOrFailed()) {
+      if (isPrimaryOrPublisher &&
+          event.state ==
+              rtc.RTCPeerConnectionState.RTCPeerConnectionStateConnected) {
+        pcState = event.state;
+      }
+
+      if (isPrimaryOrPublisher &&
+          event.state.isDisconnectedOrFailed() &&
+          pcState ==
+              rtc.RTCPeerConnectionState.RTCPeerConnectionStateConnected) {
         // trigger reconnect sequence
-        _onDisconnected(DisconnectReason.peerConnection);
+        pcState = event.state;
+        handleDisconnect(DisconnectReason.peerConnectionClosed);
```

Comment:
this seems to confuse close vs disconnected/failed. in JS, we are only triggering a full reconnect when the peer connection had gone to `closed`, but here it seems to be treating a disconnected or failed event as closed.